Dev

elismasilva opened the pull request:
### Fixes for model loading and download:

- Run download_models.py is optional now. If run, the models will be saved in the default 'models' folder within the project root path. Checkpoints will be stored in the 'checkpoints' subfolder within the 'models' directory.
- The code is already prepared so that during execution it can carry out all necessary downloads in the default 'models' folder if it does not exist in the root directory.
- The options yaml file is no longer used to reference checkpoint names.
- With the exception of the automatic download script, all code will search for model names and parameterized checkpoints in the CKPT_PTH.py file. This file will receive the default 'models' base folder in the 'model_path' variable when the 'args.ckpt_dir' argument is not provided. I created a second variable 'model_path2' which is optional if you want to use a second path for the model (e.g. kaggle temp folder). This variable is defined when the 'args.ckpt_dir2' argument is entered. If the argument is not passed, it will receive the same value as 'model_path'.

**How to test:**

**Test 1**: (Do not enter any arguments for ckpt_dir and ckpt_dir2) - Remove the 'models' folder if it exists from the project root path. Just run the code and all the necessary templates will be automatically downloaded during execution.
**Test 2**: (Enter a different path for the ckpt_dir argument) All files will be downloaded in this informed path, if they already exist, they will be used, however the existing folder structure must match the pattern of how they are originally downloaded.
**Test 3**: (Enter a different path for the ckpt_dir2 argument) The laion clip model will be downloaded to this new path.
**Test 4**: - Delete the model folder in the project root and run 'download_models.py', it should download all models back to the default folder.

**Structure model folders:**

---> **models**
| ------**checkpoints** (Here is ckpt supir and others safetensor or ckpt files)
|
|**others models folders/subfolders** (it is like model_id e.g: 'laion/CLIP-ViT-bigG-14-laion2B-39B-b160k' like subfolder)

### Fix for FP16 on Kaggle
For Kaggle you need start with follow params:

For FP16:
!python gradio_demo.py --loading_half_params --use_tile_vae --load_4bit_llava --ckpt_dir /kaggle/temp/models

For FP8:
!python gradio_demo.py --fp8  --use_tile_vae --load_4bit_llava --ckpt_dir /kaggle/temp/models

The best way to test is to use the notebook that I have already made available as there are also some dependency adjustments for the installation. For Kaggle, a separate requirements file called requirements_kaggle.txt was created.

Other changes:
FP8: --fp8 new parameter to use FP8, do not require --loading_half_params together

d8ahazard commented:
@elismasilva - A few observations, thoughts, requests after looking this over:

1. Can you please update download_model to remove the purepath requirement and just use os.path.join? Keep it consistent, remove the unnecessary bits to make it a "windows" path.

2. Why do we now have two arguments for the checkpoints folder? Is this necessary?

3. You downgraded the gradio version is requirements - is there a reason for this?

4. Your change to load_model_weights broke the test.py file. Please fix it.

5. Similar to download_model, in CKPT_PATH.py, we should be using os.path.join for os-specific paths, not assuming "/".

Sorry for a big nitpicky list, the change looks awesome and I'm excited for FP8. Just want to keep things clean and consistent.

TY!!

d8ahazard commented:
@elismasilva - Sorry, noticed one more thing:

![image](https://github.com/FurkanGozukara/SUPIR/assets/1633844/0cbbb01b-cb99-4402-bfb8-b76475b91a11)

The point behind the "ckpt_path" commandline argument was so that people could re-use their "Stable-diffusion" checkpoints directory without having to copy everything. It appears as if your changes have broken this, and now *everything* is being downloaded there.

We don't want this. IMHO, "models" should always be a static directory in the project, and if ppl want to specifically use another folder for checkpoints, they use the flag. Your logic is effectively dumping *everything* into the checkpoints folder now...

d8ahazard commented:
OK, I'm sorry, but after looking at this more and more, I see a lot of things going on that I'm not liking regarding the changes to the model stuff.

Could you please submit a new PR that only includes the 8bit stuff?

elismasilva commented:
> OK, I'm sorry, but after looking at this more and more, I see a lot of things going on that I'm not liking regarding the changes to the model stuff.
> 
> Could you please submit a new PR that only includes the 8bit stuff?

Sorry, but the purpose of the change I made to automatically download models and checkpoints was to allow any user who arrived at the repository and ran the program to run it without worrying about specifying paths or performing downloads manually. I had not understood that the checkpoint path was exclusive for the user to enter another path, but rather that it was a local cache directory. However, I corrected this so that the checkpoint path is informed again by the user. The extra path that I had created --ckpt_dir2 is optional for Kaggle where the user can enter another path to download the models. However, I have now changed the name to model_dir, which if not specified will be the static local 'models' folder. I had already made all these changes before you finished your changes, and I worked on the changes for FP8 in my parallel branch. When his changes were made available I merged them, however, Dr Furkan found a problem with the merge regarding "CR LF" which had caused changes to many files. I needed to delete my branch and take the changed files and upload them again to another branch, so at this point I can't quickly separate things from FP8 to adapt them to the old version, I think I'll have a lot of work to do that, my time It's very scarce. But I can resubmit the changes I've already made and you can test again if it satisfies you. Dr. Furkan had already anticipated these problems that you reported to me.
